sklearn/decomposition/dict_learning.py

```diff
@@ -282,9 +282,9 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
                               check_input=False,
                               verbose=verbose)
         # This ensure that dimensionality of code is always 2,
-        # consistant with the case n_jobs > 1
+        # consistent with the case n_jobs > 1
         if code.ndim == 1:
-            code = code[np.newaxis, :]
+            code = code[:, np.newaxis]
```

Comment:
You mean fix? ;) This is a bugfix, but I'm happy to make it more visible in the whatsnew. It's been a while since I wrote this, I need to look up the exact edge cases this fixes.

The controversial parts where supposed to be the complete refactoring of all the common tests and the introduction of  estimator tags. This PR fixes all the bugs that better testing finds.